test/functional/tests/io_class/test_io_class_file.py

```diff
@@ -364,53 +564,99 @@ def test_ioclass_file_size(filesystem):
+
+        ioclass_config.create_ioclass_config(
+            add_default_rule=True, ioclass_config_path=ioclass_config_path
+        )
+
+        create_directory(path=mountpoint, parents=True)
 
-    with TestRun.step("Prepare and load IO class config."):
         load_file_size_io_classes(cache, base_size)
 
-    with TestRun.step(
-        f"Prepare {filesystem.name} filesystem and mount {core.path} " f"at {mountpoint}."
-    ):
-        core.create_filesystem(filesystem)
+    with TestRun.step("Mount core"):
         core.mount(mountpoint)
         sync()
 
     with TestRun.step("Create files belonging to different IO classes (classification by writes)."):
         test_files = []
-        for size, ioclass_id in size_to_class.items():
-            occupancy_before = cache.get_io_class_statistics(
-                io_class_id=ioclass_id
-            ).usage_stats.occupancy
-            file_path = f"{mountpoint}/test_file_{size.get_value()}"
-            Dd().input("/dev/zero").output(file_path).oflag("sync").block_size(size).count(1).run()
+        for io_size, ioclass_id in size_to_class.items():
+            cache_stats = cache.get_io_class_statistics(io_class_id=ioclass_id)
+            cache_occupancy_before = cache_stats.usage_stats.occupancy
+
+            file_path = f"{mountpoint}/test_file_{io_size.get_value()}"
+            dd = (
+                Dd().input("/dev/zero").output(file_path).oflag("sync").block_size(io_size).count(1)
+            )
+            dd.run()
+
             sync()
             drop_caches(DropCachesMode.ALL)
-            occupancy_after = cache.get_io_class_statistics(
-                io_class_id=ioclass_id
-            ).usage_stats.occupancy
-            if occupancy_after != occupancy_before + size:
+
+            cache_stats = cache.get_io_class_statistics(io_class_id=ioclass_id)
+            cache_occupancy_after = cache_stats.usage_stats.occupancy
+            expected_occupancy = cache_occupancy_before + io_size
+
+            if cache_occupancy_after != cache_occupancy_before + io_size:
                 TestRun.fail(
-                    "File not cached properly!\n"
-                    f"Expected {occupancy_before + size}\n"
-                    f"Actual {occupancy_after}"
+                    "Wrong amount of cached data occurred in stats\n"
+                    f"Expected amount of cached data: {expected_occupancy}\n"
+                    f"Actual amount of cached data:{cache_occupancy_after}"
                 )
             test_files.append(File(file_path).refresh_item())
+
         sync()
         drop_caches(DropCachesMode.ALL)
 
-    with TestRun.step("Move all files to 'unclassified' IO class."):
+    with TestRun.step("Move all files to 'unclassified' IO class"):
         ioclass_config.remove_ioclass_config(ioclass_config_path)
         ioclass_config.create_ioclass_config(False, ioclass_config_path)
         ioclass_config.add_ioclass(
```

Comment:
create overwrites file contents and add_defautl_rule=True adds the default rule, so:
```suggestion
        ioclass_config.create_ioclass_config(True, ioclass_config_path)
```